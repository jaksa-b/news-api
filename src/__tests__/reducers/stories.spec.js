import storiesReducer from "../../redux/reducers/stories";
import {
  FETCH_STORIES_REQUEST,
  FETCH_STORIES_SUCCESS,
  FETCH_STORIES_FAILED,
} from "../../redux/actions/stories";

describe("Stories", () => {
  it("should return the initial state", () => {
    expect(storiesReducer(undefined, {})).toMatchSnapshot();
  });

  it(`should handle ${FETCH_STORIES_REQUEST}`, () => {
    expect(
      storiesReducer(undefined, {
        type: FETCH_STORIES_REQUEST,
        payload: { entriesId: "125670" },
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${FETCH_STORIES_SUCCESS}`, () => {
    expect(
      storiesReducer(undefined, {
        type: FETCH_STORIES_SUCCESS,
        payload: {},
      })
    ).toMatchSnapshot();
  });

  it(`should handle ${FETCH_STORIES_FAILED}`, () => {
    expect(
      storiesReducer(undefined, {
        type: FETCH_STORIES_FAILED,
        payload: {},
      })
    ).toMatchSnapshot();
  });
});
