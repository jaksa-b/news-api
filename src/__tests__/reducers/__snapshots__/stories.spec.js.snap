// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories should handle FETCH_STORIES_FAILED 1`] = `
Object {
  "data": Array [],
}
`;

exports[`Stories should handle FETCH_STORIES_REQUEST 1`] = `
Object {
  "data": Array [],
}
`;

exports[`Stories should handle FETCH_STORIES_SUCCESS 1`] = `
Object {
  "data": Array [],
}
`;

exports[`Stories should return the initial state 1`] = `
Object {
  "data": Array [],
}
`;
